### PR TITLE
Fix: Raises error when exceeding block max on metagraph

### PR DESCRIPTION
**Pull request by thealligatorking**
* Raises error now for this [Note](https://github.com/opentensor/bittensor/blob/master/bittensor/metagraph.py#L495)
* Sets block max at `300`.

### CODE REVIEW

File: bittensor/metagraph.py [hunk 1]
```diff
@@ -499,6 +499,12 @@ def sync(
 
                 subtensor = bittensor.subtensor(network='archive')
         """
+
+        if block > 300:
+            raise ValueError(
```
Comment (ifrit98) [inline bittensor/metagraph.py:509]: Can we use `bittensor.logging.error` here?  Can still raise a value error, but should be logged this way as well

**Review by ifrit98 — APPROVED** [review]
LGTM!

File: bittensor/metagraph.py (hunk 1 above)
Comment (thealligatorking) [reply bittensor/metagraph.py:509]: Yep!

Comment (ifrit98) [reply bittensor/metagraph.py:509]: Looks like we're getting test errors now

File: tests/unit_tests/test_metagraph.py [hunk 2]
```diff
@@ -133,7 +133,7 @@ def mock_subtensor():
 # Mocking the metagraph instance for testing purposes
 @pytest.fixture
 def metagraph_instance():
-    metagraph = Metagraph(netuid=1, sync=False)
+    metagraph = Metagraph(netuid=1337, sync=False)
```
Comment (ifrit98) [inline tests/unit_tests/test_metagraph.py:136]: :)

**Pull request merged**